ln-resource-mgr/src/lib.rs

```diff
@@ -203,7 +202,20 @@ impl AllocationCheck {
                         return Ok(ResourceBucketType::Congestion);
                     }
 
-                    Err(FailureReason::NoReputation)
+                    match reputation_check {
```

Comment:
Let's add a comment here explaining that if we've got outgoing reputation in play, we're required to drop endorsed htlcs to peers without reputation to protect against the unknown peer possibly damaging our reputation with our upstream peers + if we're only looking at incoming, we're not at risk because we're not held accountable for the behavior of downstream nodes.